aws-crt-http/HttpConnection.cs

```diff
@@ -28,39 +30,135 @@ public enum OutgoingBodyStreamState
         Done = 1,
     }
 
-    public delegate void OnConnectionSetup(int errorCode);
-    public delegate void OnConnectionShutdown(int errorCode);
-    public delegate OutgoingBodyStreamState OnStreamOutgoingBody(HttpClientStream stream, byte[] buffer, out UInt64 bytesWritten);
-    public delegate void OnIncomingHeaders(HttpClientStream stream, HttpHeader[] headers);
-    public delegate void OnIncomingHeaderBlockDone(HttpClientStream stream, bool hasBody);
-    public delegate void OnIncomingBody(HttpClientStream stream, byte[] data, ref UInt64 windowSize);
-    public delegate void OnStreamComplete(HttpClientStream stream, int errorCode);
-
-    internal delegate int OnStreamOutgoingBodyNative(
-        [In, Out, MarshalAs(UnmanagedType.LPArray, SizeParamIndex=1)] byte[] buffer, 
-        UInt64 size,
-        out UInt64 bytesWritten);
-    internal delegate void OnIncomingHeadersNative(
-        Int32 responseCode,
-        [In, MarshalAs(UnmanagedType.LPArray, SizeParamIndex=2)] HttpHeader[] headers, 
-        UInt32 count);
-    internal delegate void OnIncomingHeaderBlockDoneNative(bool hasBody);
-    internal delegate void OnIncomingBodyNative(
-        [In, MarshalAs(UnmanagedType.LPArray, SizeParamIndex=1)] byte[] buffer, 
-        UInt64 size,
-        ref UInt64 windowSize);
-    internal delegate void OnStreamCompleteNative(int errorCode);
+    public class StreamResult
+    {
+        public int ErrorCode { get; internal set; }
+
+        internal StreamResult(int errorCode)
+        {
+            ErrorCode = errorCode;
+        }
+    }
+
+    public abstract class HttpClientStreamEventArgs : EventArgs
+    {
+        public HttpClientStream Stream { get; private set; }
+
+        public HttpClientStreamEventArgs(HttpClientStream stream)

```

Comment:
if the setter is private, should the constructor be marked internal?

Reply:
Good call, fixed.